### PR TITLE
Add "Order of Nested Selectors" Section

### PR DESCRIPTION
**Pull request by spaceninja**
This PR adds an "Order of Nested Selectors" section that is copied from the [CSS Styleguide](https://github.com/spaceninja/css-standards#order-of-properties) I wrote for my last job. It tries to emphasize a logical structure to nested selectors: Included styles, then regular styles, then modifiers, then children.

Feedback welcome!

### CODE REVIEW

File: css/README.md [hunk 1]
```diff
@@ -239,19 +239,126 @@ These rules were adapted from [CSS Guidelines]. This is an example of how declar
...
+
+2. #### Alphabetized Regular Styles
+	1. Adding regular styles after the `@extend` and `@include` statements allows us to override those properties, if needed. Be sure to leave a comment if a declaration's only purpose is to override included styles. 
+    
+	1. Alphabetizing is helpful for a large team because it standardizes location of properties (See "Property Sorting" below).
+
+3. #### Modifiers
+	1. Pseudo-classes (`:hover`), attribute selectors (`&[type="input"]`), and state classes (`&.is-active`)
+		* These directly modify the parent selector so we declare them before other nested selectors.
+        * Most BEM modifier classes don't need to be nested. However, some generic state class names like `is-active` that would otherwise be too non-specific to live in the global namespace, should be nested here.
```
Comment (derekshirk) [inline css/README.md:276]: It's hard for me to tell, is the indendation here as you intended?

File: css/README.md [hunk 2]
```diff
@@ -239,19 +239,126 @@ These rules were adapted from [CSS Guidelines]. This is an example of how declar
...
+		* However, if you have a set of variables that will only be used in a certain partial, declare them at the top of that file and prefix their names to avoid confusion in the global namespace. e.g., `$m_module_name_height`.
+
+	1. `@extend` statements
+		* Avoid extends in favor of mixins. Extends offer no benefits over mixins, and can cause [specificity problems](http://csswizardry.com/2016/02/mixins-better-for-performance/).
+
+	1. `@include` statements
+		* Knowing right off the bat that this class inherits another whole set of rules from elsewhere is good. Another benefit is that overriding styles for that inherited set of rules becomes much easier.
+
+2. #### Alphabetized Regular Styles
+	1. Adding regular styles after the `@extend` and `@include` statements allows us to override those properties, if needed. Be sure to leave a comment if a declaration's only purpose is to override included styles.
```
Comment (derekshirk) [inline css/README.md:269]: Pardon my naivety,  but is this statement related to "alphabetizing regular styles"?

File: css/README.md [hunk 3]
```diff
@@ -239,19 +239,126 @@ These rules were adapted from [CSS Guidelines]. This is an example of how declar
...
+	2. Nested selectors
+		* As a rule, **if a selector will work without it being nested then do not nest it.**
+		* There shouldn’t be many nested selectors. Our naming convention means that we don’t need to nest selectors for namespacing. Before you nest a selector, consider if it would be better as a child class or modifier class.
+		* Any nested selectors could contain their own nested pseudo-classes, pseudo-elements and media queries, following the rules above.
+
+```sass
+.foo {
+    @include font_ui(2.0); // 1.iii
+    line-height: 1;        // override font_ui line-height // 2.i
+    width: 100%;           // 2.ii
```
Comment (derekshirk) [inline css/README.md:298]: I noticed above there are numerical comments styled like this: 

`/* 1 */` 

and here comments are styled as:

`//1.iii 

Does that matter?

Also, I was confused by double, double slashes in this comment: 

`// override font_ui line-height // 2.i`

Is that intentional?

File: css/README.md [hunk 4]
```diff
@@ -239,19 +239,126 @@ These rules were adapted from [CSS Guidelines]. This is an example of how declar
...
+```
+
 ### Property Sorting
 
 Until recently, we followed Guy Routledge's [Outside In] method of property sorting. Over time, we discovered most property sorting methodologies require maintenance as new properties (for example, `flex-*` or `grid-*`) are introduced. This makes them inherently fragile and difficult to enforce.
 
 We now recommend properties are sorted alphabetically. This technique will scale to accommodate any new properties, and requires no learning curve for new project contributors.
 
+* You can quickly alphabetize declarations in Sublime Text by selecting several lines of code and pressing F5.
+* You can quickly alphabetize declarations in VS Code by selecting several lines of code and using the "Sort Lines Ascending" command. This has no keyboard shortcut by default, so you can either use the command pallete (CMD+SHIFT+P) or assign a keyboard shortcut.
```
Comment (derekshirk) [inline css/README.md:360]: TIL! Thank you.

File: css/README.md (hunk 2 above)
Comment (spaceninja) [reply css/README.md:269]: Yes — this is explaining why alphabetized regular styles come after `@extend` and `@include` statements.

File: css/README.md (hunk 3 above)
Comment (spaceninja) [reply css/README.md:298]: Yeah, the `1.iii` references the way the ordered list numbrers are rendered in the document. When you view the rendered markdown, this makes it easier to tell which item it's referring to.

and no, the double comment was a mistake, I'll fix that.

File: css/README.md (hunk 1 above)
Comment (spaceninja) [reply css/README.md:276]: Nope, my indenting got messed up. I'll fix it.
